Article/Article.js

```diff
@@ -1,26 +1,131 @@
+    }
+    else if (!this.domElement.classList.contains('article-open')) {
+      this.expandButton.innerText = "EXPAND";
+    }
   }
 }
 
-// START HERE: Select all classes named ".article" and assign that value to the articles variable
-let articles;
+// START HERE: Select all classes named ".article" and 
+//assign that value to the articles variable
+let articles = document.querySelectorAll('.article');
+
+// Use .map() to iterate over the articles array and 
+//create a new instance of Article by passing in 
+//each article as a parameter to the constructor.
+
+articles = Array.from(articles).map(article => new Article(article));
+
+//    FORM BOX EXPERIMENT  \\
+
+let formBox = document.querySelector('.form-box');
+
+formBox.addEventListener('click', function () {
+  formBox.classList.toggle('reg-form');
+  formBox.classList.remove('btn-form');
+}, { once: true })
+
+let publishBtn = document.querySelector('.publish-btn');
+
+let articleContainer = document.querySelector('.articles');
+
+
+
+
+let titleInput = document.querySelector('#title').value;
+let dateInput = null;
+let paraInput = document.querySelector('textarea').value;
 
-// Use .map() to iterate over the articles array and create a new instance of Article by passing in each article as a parameter to the constructor.
-articles = Array.from(articles);
 
+
+
+class Articles {
+  constructor(props) {
+
+    this.props = props;
+    this.artBody = document.createElement('div');
+    this.h2 = document.createElement('h2');
+    this.date = document.createElement('p');
+    this.p = document.createElement('p');
+    this.expandBtn = document.createElement('span');
+
+    this.artBody.classList.add('article');
+    this.date.classList.add('date');
+    this.expandBtn.classList.add('expandButton');
+    this.expandBtn.innerText ='EXPAND';
+
+    this.artBody.appendChild(this.h2);
+    this.artBody.appendChild(this.date);
+    this.artBody.appendChild(this.p);
+    this.artBody.appendChild(this.expandBtn);
+    if (this.props['title']) {
+      this.h2.innerText = this.props['title'];
+    }
+    if (this.props['date']) {
+      this.date.innerText = this.props['date'];
+    }
+    if (this.props['para']) {
+      this.p.innerText = this.props['para'];
+    }
+
+    this.expandBtn.addEventListener('click', this.expandArticle.bind(this));
+
+  }
+    expandArticle() {
+      this.artBody.classList.toggle('article-open');
+      if (this.artBody.classList.contains('article-open')) {
+        this.expandBtn.innerText = "CLOSE";
+      }
+      else if (!this.artBody.classList.contains('article-open')) {
+        this.expandBtn.innerText = "EXPAND";
+
+      }
+    }
+
+  }
+
+  publishBtn.addEventListener('click', function () {
+    let artObj = {
+      'title': titleInput ,
+      'date': dateInput , 
+      'para': paraInput}
+  
+    let newArt = new Articles(artObj).innerHTML;
+    articleContainer.append(new Articles(artObj).artBody.innerHTML);
```

Comment:
here, you are creating a newArticle component, accessing the div element, then accessing that innerHTML, which is a string, and then you append that string to articleContainer. But you don't append strings, you append *nodes*. So this may work:
```
articleContainer.append(new Articles(artObj).
```